src/components/single-country/SingleCountry.jsx

```diff
@@ -0,0 +1,71 @@
+import { useState } from 'react';
+import './single-country.css';
+import WorldMap from '../world-map/WorldMap';
+import CountryCard from '../country-card/CountryCard';
+import Button from '../button/Button';
+import roundToMillion from '../../helpers/roundToMillion';
+
+const SingleCountry = function ({ userChoice, countryData, setSingleCountryData, fetchData }) {
+    const [query, setQuery] = useState('');
+    const [errorMessage, setErrorMessage] = useState('');
+
+    let data;
+    // Checks if countryData.data exists and has at least 1 element before trying to use it
+    if (countryData && countryData.data && countryData.data.length > 0) {
+        data = countryData.data[0];
+    }
```

Comment:
```suggestion
    const data = countryData?.data?.[0];
```
Dit zou hetzelfde resultaat moeten geven

Reply:
Tested and adopted feedback.